synapse/appservice/api.py

```diff
@@ -202,8 +198,7 @@ def _get():
         key = (service.id, protocol)
         return self.protocol_meta_cache.wrap(key, _get)
 
-    @defer.inlineCallbacks
-    def push_bulk(self, service, events, txn_id=None):
+    async def push_bulk(self, service, events, txn_id=None):
```

Comment:
https://github.com/matrix-org/synapse/blob/70e9a40ddcd698ec45377d8520d6c9dc842164ea/synapse/appservice/__init__.py#L43-L48 should probably be updated as well.

Reply:
Good catch! 👍